lib/test/unit/assertions.rb

```diff
@@ -998,9 +1003,16 @@ def _assert_in_delta_message(expected_float, actual_float, delta,
 EOT
         end
         arguments = [expected_float, delta, actual_float]
-        normalized_expected = expected_float.to_f
-        normalized_actual = actual_float.to_f
-        normalized_delta = delta.to_f
+        if %i(+ - <).all? {|sym| expected_float.respond_to?(sym) } &&
+            %i(< <=).all? {|sym| actual_float.respond_to?(sym) }
```

Comment:
@kou How about this?  Is it acceptable the difference in type checking way from `assert_in_delta`?

Reply:
I want to avoid it if possible. How about this?

```diff
diff --git a/lib/test/unit/assertions.rb b/lib/test/unit/assertions.rb
index 22f1da1..37a7214 100644
--- a/lib/test/unit/assertions.rb
+++ b/lib/test/unit/assertions.rb
@@ -929,12 +929,27 @@ def assert_in_delta
           _assert_in_delta_validate_arguments(expected_float,
                                               actual_float,
                                               delta)
-          full_message = _assert_in_delta_message(expected_float,
-                                                  actual_float,
-                                                  delta,
-                                                  message)
+          begin
+            pass = (expected_float - actual_float).abs <= delta
+            full_message = _assert_in_delta_message(expected_float,
+                                                    expected_float,
+                                                    actual_float,
+                                                    actual_float,
+                                                    delta,
+                                                    delta,
+                                                    message)
+          rescue
+            pass = (expected_float.to_f - actual_float.to_f).abs <= delta.to_f
+            full_message = _assert_in_delta_message(expected_float,
+                                                    normalized_expected_float,
+                                                    actual_float,
+                                                    normalized_actual_float,
+                                                    delta,
+                                                    normalized_delta,
+                                                    message)
+          end
           assert_block(full_message) do
-            (expected_float.to_f - actual_float.to_f).abs <= delta.to_f
+            pass
           end
         end
       end
```

Reply:
I like this. It's easier than my approach.